apiserver/common/controllerconfig.go

```diff
@@ -76,15 +77,44 @@ func (s *controllerStateShim) ControllerInfo(modelUUID string) (addrs []string,
 		return StateControllerInfo(s.State)
 	}
 
-	// Now check any external controllers.
 	ec := state.NewExternalControllers(s.State)
 	info, err := ec.ControllerForModel(modelUUID)
+
+	// the model may have been migrated, if that is the case we try to get the information.
+	// If that is true we try to save the migration controller as an external controller.
+	if errors.IsNotFound(err) {
+		migration, err := s.State.LatestRemovedModelMigrationByUUID(modelUUID)
+		if err != nil {
+			return handleControllerInfoErr(err)
+		}
+		target, err := migration.TargetInfo()
+		if err != nil {
+			return handleControllerInfoErr(err)
+
+		}
+		logger.Debugf("found migrated model on another controller. Saving it")
+		_, err = ec.Save(crossmodel.ControllerInfo{
+			ControllerTag: target.ControllerTag,
+			Alias:         target.ControllerAlias,
+			Addrs:         target.Addrs,
+			CACert:        target.CACert,
+		}, modelUUID)
+		if err != nil {
+			return handleControllerInfoErr(err)
+		}
+		return target.Addrs, target.CACert, nil
+	}
+
 	if err != nil {
-		return nil, "", errors.Trace(err)
+		return handleControllerInfoErr(err)
```

Comment:
I don't know what this gives you?

Reply:
for me, it is mostly usability with the ide. So if I need to return an err again I can just start the functionname and the IDE auto completed for me, instead of me typing those words. But I have no hard feelings about that. I can change it.

Reply:
It means another function call I have to lookup, that just wraps the error. It inhibits readability for little gain imo.

Reply:
Fair point. For me, either version works for me. I can change that with the other review points (when given)